server/index.js

```diff
@@ -32,22 +32,43 @@ app.use('/api/users', userRoutes)
 app.use('/api/posts', postRoutes)
 app.use('/api/upload', uploadRoute)
 
-mongoose.connect(process.env.MONGO_URI, { useNewUrlParser: true, useUnifiedTopology: true }, () => {
-  console.log("Connected to database")
+const dbConnect = async () => new Promise((resolve) => {
+  mongoose.connect(process.env.MONGO_URI, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    serverSelectionTimeoutMS: 10000,
+  }, (err) => {
+    if (err) {
+      throw err
+    }
+
+    console.log(`🛢[database]: Connected to database at ${process.env.MONGO_URI}`)
+    resolve()
+  })
 })
 
 app.use((err, req, res, next) => {
   const status = err.status || 500
   const message = err.message || "Something went wrong!"
+  const log = status >= 500 ? console.error : console.warn;
+
+  log(err);
```

Comment:
Log server-side errors to the console too, which is where I'm used to looking for such things!